src/SourceControl-General/WonderBackend.ts

```diff
@@ -30,12 +30,12 @@ function GetOrAddWonderBackendProject(
  * @param  {type} wonderBackend: WonderBackend the backend to sync
  * @return {type}                              whether or not this succeeded
  */
-export function SyncProjectWithWonder(
+export async function SyncProjectWithWonder(
   project: Project,
   wonderBackend: WonderBackend
-): boolean {
+): Promise<Boolean> {
   // update the project so we get the latest state
-  project.sync();
+  await project.synchronize();
```

Comment:
any reason for this to be outside of the `try/catch` ? If the exception occurs, it should be caught in that block.